src/components/MapView/Legends/index.tsx

```diff
@@ -76,35 +89,113 @@ function Legends({ classes, layers }: LegendsProps) {
 }
 
 // Children here is legendText
-function LegendItem({ classes, title, legend, children }: LegendItemProps) {
+function LegendItem({
+  classes,
+  id,
+  title,
+  legend,
+  type,
+  opacity: initialOpacity,
+  children,
+}: LegendItemProps) {
+  const map = useSelector(mapSelector);
+  const [opacity, setOpacityValue] = useState<number | number[]>(
+    initialOpacity || 0,
+  );
+
+  const [toggle, setToggle] = useState(true);
+  const toggleChecked = (prev: Boolean) => {
```

Comment:
Was the captial `Boolean` instead of lowercase `boolean` intentional?

Reply:
sorry did't realize this one, but i removed the lines related to toggle feature and will submit it to another pull request